src/index.ts

```diff
@@ -134,17 +129,23 @@ function requestToFetchOptions(reqOpts: Options) {
     uri = uri + '?' + params;
   }
 
-  const isHttp = /^http:\/\//.test(uri);
+  const isHttp = uri.startsWith('http://');
   const proxy =
     reqOpts.proxy ||
     process.env.HTTP_PROXY ||
     process.env.http_proxy ||
     process.env.HTTPS_PROXY ||
     process.env.https_proxy;
   if (proxy) {
-    options.agent = isHttp
-      ? new HttpProxyAgent(proxy)
-      : new HttpsProxyAgent(proxy);
+    if (isHttp) {
+      // tslint:disable-next-line variable-name
+      const HttpProxyAgent = require('http-proxy-agent');
```

Comment:
Apologies - I should have called this out.  Could you cache the value and only do the `require` once?  Otherwise this gets called in every HTTP request

Reply:
in air conversation, shared the arcane Node fact that is `Module._cache`.